CustomScript/customscript.py

```diff
@@ -103,37 +103,81 @@ def enable(hutil):
+    for retry in range(0, tries):
         try:
             download_files(hutil)
             break
         except Exception, e:
-            hutil.error(("Failed to download files, "
-                         "retry={0}, maxRetry={1}").format(retry, maxRetry))
-            if retry != maxRetry:
-                hutil.log("Sleep 10 seconds")
-                time.sleep(10)
-            else:
-                error_msg = "Failed to download files"
-                waagent.AddExtensionEvent(name=ExtensionShortName,
-                                          op=DownloadOp,
-                                          isSuccess=False,
-                                          version=Version,
-                                          message="(01100)"+error_msg)
-                raise
+            hutil.error("Failed to download files, retry.")
+            hutil.log("Sleep {0} seconds".format(wait))
+            time.sleep(wait)
+    else:
+        error_msg = "Failed to download files"
+        hutil.error(error_msg)
+        waagent.AddExtensionEvent(name=ExtensionShortName,
+                                  op=DownloadOp,
+                                  isSuccess=False,
+                                  version=Version,
+                                  message="(01100)"+error_msg)
+        raise
+    msg = "Succeeded to download files, retry count={0}".format(retry)
+    hutil.log(msg)
     waagent.AddExtensionEvent(name=ExtensionShortName,
                               op=DownloadOp,
                               isSuccess=True,
                               version=Version,
-                              message="(01303)Succeeded to download files")
+                              message="(01303)"+msg)
+
+    # The internal DNS needs some time to be ready.
+    # Wait and retry to check if there is time in retry window.
+    is_idns_ready = check_idns()
+    while (not is_idns_ready):
+        if retry == tries:
+            error_msg = "Internal DNS is not ready, ignore it."
+            hutil.error(error_msg)
+            waagent.AddExtensionEvent(name=ExtensionShortName,
+                                      op="CheckIDNS",
+                                      isSuccess=False,
+                                      version=Version,
+                                      message="(01306)"+error_msg)
+            break
+        hutil.error("Internal DNS is not ready, retry to check.")
+        hutil.log("Sleep {0} seconds".format(wait))
+        time.sleep(wait)
+        is_idns_ready = check_idns()
+        retry = retry + 1
+    else:
+        msg = "Internal DNS is ready"
+        hutil.log(msg)
+        waagent.AddExtensionEvent(name=ExtensionShortName,
+                                  op="CheckIDNS",
+                                  isSuccess=True,
+                                  version=Version,
+                                  message="(01306)"+msg)
```

Comment:
Like what we synced offline, please add telemetry for both downloading and iDNS retry count.